tests/functional/nix-profile.sh

```diff
@@ -104,34 +104,35 @@ nix profile upgrade 0
 nix profile history | grep "packages.$system.default: 1.0, 1.0-man -> 3.0, 3.0-man"
 
 # Test new install of CA package.
-nix profile remove 0
+nix profile remove flake1
 printf 4.0 > $flake1Dir/version
 printf Utrecht > $flake1Dir/who
 nix profile install $flake1Dir
 [[ $($TEST_HOME/.nix-profile/bin/hello) = "Hello Utrecht" ]]
 [[ $(nix path-info --json $(realpath $TEST_HOME/.nix-profile/bin/hello) | jq -r .[].ca) =~ fixed:r:sha256: ]]
 
 # Override the outputs.
-nix profile remove 0 1
+nix profile remove simple flake1
 nix profile install "$flake1Dir^*"
 [[ $($TEST_HOME/.nix-profile/bin/hello) = "Hello Utrecht" ]]
 [ -e $TEST_HOME/.nix-profile/share/man ]
 [ -e $TEST_HOME/.nix-profile/include ]
 
 printf Nix > $flake1Dir/who
-nix profile upgrade 0
+nix profile list
+nix profile upgrade flake1
 [[ $($TEST_HOME/.nix-profile/bin/hello) = "Hello Nix" ]]
 [ -e $TEST_HOME/.nix-profile/share/man ]
 [ -e $TEST_HOME/.nix-profile/include ]
 
-nix profile remove 0
+nix profile remove flake1
 nix profile install "$flake1Dir^man"
 (! [ -e $TEST_HOME/.nix-profile/bin/hello ])
 [ -e $TEST_HOME/.nix-profile/share/man ]
 (! [ -e $TEST_HOME/.nix-profile/include ])
 
 # test priority
-nix profile remove 0
+nix profile remove flake1
 
 # Make another flake.
```

Comment:
Could we get an integration test confirming the behavior when two flakes have matching names/heuristics?

Reply:
Good point, I'll try to make one up :+1: The more thorough an integration test, the more potential issues it can uncover. I'd rather test a feature at multiple levels.

Reply:
Ok I added one that basically copies a simple nix file to a different directory and installs it from there. Works like a charm.